ide/src/extension.ts

```diff
@@ -1,28 +1,48 @@
 import * as vscode from "vscode";
 import _ from "lodash";
-import { CallFlowistry, log, show_error } from "./vsc_utils";
+import {
+  CallFlowistry,
+  FlowistryErrorDocument,
+  last_error,
+  log,
+  show_error,
+} from "./vsc_utils";
 
 import { decompose } from "./decompose";
-import { focus, focus_mark, focus_unmark, focus_select } from "./focus";
+import { FocusMode } from "./focus";
 import { setup } from "./setup";
 
 import "./app.scss";
 
+export let flowistry_status_bar_item: vscode.StatusBarItem;
+export const tdcp = new FlowistryErrorDocument();
```

Comment:
Still might need some global state 😔

The text document provider can only be defined once to use the same uri scheme and the status bar item is imported/modified in a couple distinct functions. Mentioned this in the PR description, but this could be solved by propagating/drilling these into each function they're used (although this is arguably worse than making them globally accessible).